Script for generating image files

create cli for generating image file from spec

basically just wiring `vl2vg` and vega's `vg2png` and `vg2svg` should work

I'm stuck with the error that ```Cannot find module 'canvas'```
I'm new to node and NPM. But as I know, ``vega`` includes ``canvas`` and ``vega-lite`` includes ``vega``. So I think that we don't have to include ``canvas`` again.   
Could you( @kanitw ) comment about this? 

```
module.js:339
    throw err;
    ^

Error: Cannot find module 'canvas'
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.Module._load (module.js:287:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.instance (/Users/younghoonkim/Desktop/dev/vega-lite/node_modules/vega/node_modules/vega-scenegraph/src/util/canvas.js:11:18)
    at CanvasRenderer.prototype.initialize (/Users/younghoonkim/Desktop/dev/vega-lite/node_modules/vega/node_modules/vega-scenegraph/src/render/canvas/CanvasRenderer.js:20:25)
    at View.prototype.initialize (/Users/younghoonkim/Desktop/dev/vega-lite/node_modules/vega/src/core/HeadlessView.js:57:6)
    at View.prototype.renderer (/Users/younghoonkim/Desktop/dev/vega-lite/node_modules/vega/src/core/View.js:235:10)
    at View.prototype.renderer (/Users/younghoonkim/Desktop/dev/vega-lite/node_modules/vega/src/core/HeadlessView.js:16:34)
    at /Users/younghoonkim/Desktop/dev/vega-lite/node_modules/vega/src/core/View.js:435:8
```

--- a/bin/vl2png
+++ b/bin/vl2png
@@ -1,0 +1,66 @@
+#!/usr/bin/env node
+// Render a Vega specification to PNG, using node canvas
+
+var helpText =
+  'Render a Vega-lite specification to PNG.\n\n' +
+  'Usage: vl2png vega_lite_json_file [output_png_file]\n' +
+  '  If output_png_file is not provided, writes to stdout.\n\n' +
+  'To load data, you may need to set a base directory:\n' +
+  '  For web retrieval, use `-b http://host/data/`. \n' +
+  '  For files, use `-b file:///dir/data/` (absolute) or `-b data/` (relative).';
+
+// import required libraries
+var path = require('path'),
+    fs = require('fs'),
+    vl = require('../vega-lite.js'),
+    vg = require('vega');
+
+// arguments
+var args = require('yargs')
+  .usage(helpText)
+  .demand(1)
+  .string('b').alias('b', 'base')
+  .describe('b', 'Base directory for data loading.')
+  .argv;
+
+// set baseURL if provided on command line
+var base = 'file://' + process.cwd() + path.sep;
+if (args.b) {
+  // if no protocol, assume files, relative to current dir
+  base = /^[A-Za-z]+\:\/\//.test(args.b) ? args.b + path.sep
+    : 'file://' + process.cwd() + path.sep + args.b + path.sep;
+}
+vg.config.load.baseURL = base;
+
+// input / output files
+var specFile = args._[0],
+    outputFile = args._[1] || null;
+
+// load spec, render to png
+fs.readFile(specFile, 'utf8', function(err, text) {
+  if (err) throw err;
+  var vlSpec = JSON.parse(text);
+  var vgSpec = vl.compile(vlSpec).spec;
+  render(vgSpec);
+});
+
+// ---
+
+function writePNG(canvas, file) {
+  var out = file ? fs.createWriteStream(file) : process.stdout;
+  var stream = canvas.createPNGStream();
+  stream.on('data', function(chunk) { out.write(chunk); });
+}
+
+function render(spec) {
+  vg.parse.spec(spec, function(err, chart) {
+    if (err) { throw err; }
+
+    var view = chart({ renderer: 'canvas' })
+      .update();
+
+    view.canvasAsync(function(canvas) {
+      writePNG(canvas, outputFile);
+    });
+  });
+}

--- a/bin/vl2svg
+++ b/bin/vl2svg
@@ -1,0 +1,76 @@
+#!/usr/bin/env node
+// Render a Vega specification to SVG
+
+var helpText =
+  'Render a Vega specification to SVG.\n\n' +
+  'Usage: vg2svg vega_lite_json_file [output_png_file]\n' +
+  '  If output_svg_file is not provided, writes to stdout.\n\n' +
+  'To load data, you may need to set a base directory:\n' +
+  '  For web retrieval, use `-b http://host/data/`. \n' +
+  '  For files, use `-b file:///dir/data/` (absolute) or `-b data/` (relative).';
+
+var svgHeader =
+  '<?xml version="1.0" encoding="utf-8"?>\n' +
+  '<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" ' +
+  '"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">\n';
+
+// import required libraries
+// import required libraries
+var path = require('path'),
+    fs = require('fs'),
+    vl = require('../vega-lite.js'),
+    vg = require('vega');
+
+// arguments
+var args = require('yargs')
+  .usage(helpText)
+  .demand(1)
+  .string('b').alias('b', 'base')
+  .describe('b', 'Base directory for data loading.')
+  .boolean('h').alias('h', 'header')
+  .describe('h', 'Include XML header and SVG doctype.')
+  .argv;
+
+// set baseURL if provided on command line
+var base = 'file://' + process.cwd() + path.sep;
+if (args.b) {
+  // if no protocol, assume files, relative to current dir
+  base = /^[A-Za-z]+\:\/\//.test(args.b) ? args.b + path.sep
+    : 'file://' + process.cwd() + path.sep + args.b + path.sep;
+}
+vg.config.load.baseURL = base;
+
+// input / output files
+var header = args.h ? svgHeader : '',
+    specFile = args._[0],
+    outputFile = args._[1] || null;
+
+// load spec, render to svg
+fs.readFile(specFile, 'utf8', function(err, text) {
+  if (err) throw err;
+  var vlSpec = JSON.parse(text);
+  var vgSpec = vl.compile(vlSpec).spec;
+  render(vgSpec);
+});
+
+// ---
+
+function writeSVG(svg, file) {
+  svg = header + svg;
+  if (file) { // write to file
+    fs.writeFile(file, svg, function(err) { if (err) throw err; });
+  } else {    // write to stdout
+    process.stdout.write(svg);
+  }
+}
+
+function render(spec) {
+  vg.parse.spec(spec, function(err, chart) {
+    if (err) { throw err; }
+
+    var view = chart({ renderer: 'svg' })
+      .update();
+
+    writeSVG(view.svg(), outputFile);
+  });
+}

